lib/src/test/java/com/thenewboston/api/bankapi/datasource/BankDataSourceTest.kt

```diff
@@ -358,6 +367,22 @@ class BankDataSourceTest {
                         response.cause?.message shouldBe "An error occurred while sending invalid block"
                     }
                 }
+
+                @Test
+                fun `should return error outcome when sending block`() {
+                    runBlockingTest {
+                        // given
+                        val request = Mocks.postBlockRequest()
+
+                        // when
+                        val response = bankDataSource.sendBlock(request)
+
+                        // then
+                        check(response is Outcome.Error)
+                        response.cause should beInstanceOf<IOException>()
+                        response.cause?.message shouldBe "An error occurred while sending the block"
+                    }
+                }
```

Comment:
We need another test to cover this error as well:
` response.balanceKey.isBlank() -> {
                val message = "Received invalid response when sending block with balance key: ${request.message.balanceKey}"
                Outcome.Error(message, IOException())
            }`

Reply:
This error case above can go in the @Display("Given empty or invalid response body..") inner class.

Reply:
resolved